### PR TITLE
Small refactor

### PR DESCRIPTION
**Pull request by ryanmrichard**
This PR:
- adds PyBerny to the build system
- updates the build system so that it can "build" the tests (doesn't work yet though, vide infra)
- adds YAML needed for using GitHub actions to test the code.
- removes dependence on OpenMOPAC (we don't want people to be forced to used this)
- creates a unit test (which doesn't work yet)

### CODE REVIEW

**Comment by ryanmrichard** [comment]
@jlheflin I'll leave it up to you if you want to read through these changes or just merge them. If you do want to read through them I'd focus on the ones to `pybernyopt.py` and the files needed to add the unit tests (files under `tests/python/unit_tests/test_optimizers`). Let me know if you have questions.

Once you've merged these we should talk about how to get an energy and gradient for unit testing so we can merge this into master.

**Review by jlheflin — APPROVED** [review]

**Pull request merged**
